### ISSUE TITLE
Basic Keras model underperforming against Scikit-Learn MLPRegressor model

### ISSUE THREAD

**Issue by jhogg11** [tensorflow/tensorflow#56599]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

binary

### Tensorflow Version

2.9.0

### Custom Code

Yes

### OS Platform and Distribution

macOS Monterey

### Python version

3.9

### Current Behaviour?

I've experimented with sklearn's MLPRegressor class and have seen that it does fairly well for the dataset I'm looking at without much tuning. However, I'd like to be able to build out a more complex model in Tensorflow/Keras using a split LSTM and Dense network.

To fulfill that end, I'm trying to first replicate the performance of MLPRegressor in Tensorflow for a very basic architecture, but struggling so far. Here's an attempt to create identical models with each. The parameters in the TF implementation are intended to be based on the MLPRegressor documentation, including certain default values.

Running the code below, there are two noticeable observations:
- The loss during training is roughly half for MLPRegressor versus the TF model. This is also what I've observed on the real dataset.
- The final MSE of the predictions on the training set is always lower for MLPRegressor (note: I'm not sure if the random seeds have the same effect on both models, but if you run it a few times, you should see this).

Just a note - I've posted this on StackOverflow, Reddit, and sklearn's GitHub without any substantive responses, even though the code should be fully reproducible.  This is kind of a last resort, but I'm hoping that someone here might be interested in why TF could potentially underperform against sklearn.

### Standalone code to reproduce the issue

```shell
import numpy as np
import matplotlib.pyplot as plt

from tensorflow.keras.layers import Input, Dense
from tensorflow.keras.models import Model
from tensorflow.keras.wrappers.scikit_learn import KerasRegressor
from tensorflow.keras.regularizers import L2
from tensorflow.random import set_seed

from sklearn.neural_network import MLPRegressor
from sklearn.pipeline import Pipeline
from sklearn.compose import TransformedTargetRegressor
from sklearn.preprocessing import StandardScaler

from sklearn.datasets import make_regression
from sklearn.metrics import mean_squared_error


all_sk = []
all_tf = []

for _ in range(100):

    X, y = make_regression(n_samples=10000, n_features=20, n_informative=10, n_targets=1)
    # y = StandardScaler().fit_transform(y[:,None])[:,0]
    
    use_scaling = True
    seed = np.random.randint(0,1000)
    
    
    def simple_tf_model():
        set_seed(seed)
        dense_input = Input(shape=(X.shape[1],))
        dense = Dense(100, activation="relu", kernel_regularizer=L2(l2=0.0001))(dense_input)
        dense = Dense(1, activation="linear", kernel_regularizer=L2(l2=0.0001))(dense)     
        tf_model = Model(inputs=[dense_input], outputs=dense)     
        tf_model.compile(loss="mse",optimizer="adam")
        return tf_model
    
    
    sk_model = MLPRegressor(max_iter=5, hidden_layer_sizes=(100,), batch_size=200, random_state=seed, verbose=1)
    tf_model = KerasRegressor(build_fn=simple_tf_model, batch_size=200, epochs=5, validation_split=0.1)
    
    if use_scaling:
        sk_pipeline = Pipeline([('scaler', StandardScaler()), ('model', sk_model)])
        sk_model = TransformedTargetRegressor(regressor=sk_pipeline, transformer=StandardScaler()) 
        
        tf_pipeline = Pipeline([('scaler', StandardScaler()), ('model', tf_model)])
        tf_model = TransformedTargetRegressor(regressor=tf_pipeline, transformer=StandardScaler()) 
        

    sk_model.fit(X,y)
    tf_model.fit(X,y)
    
    sk_preds = sk_model.predict(X)
    tf_preds = tf_model.predict(X)
    
    
    def get_mse(preds, name):
        print(name, mean_squared_error(preds, y))
        
        if name == "SK":
            all_sk.append(mean_squared_error(preds, y))
        else:
            all_tf.append(mean_squared_error(preds, y))
    
    get_mse(sk_preds, "SK")
    get_mse(tf_preds, "TF")
    

sk_arr = np.array(all_sk)
tf_arr = np.array(all_tf)

print(sk_arr.mean()). # 350.0151514048654
print(tf_arr.mean()) # 382.19699899150226

```

### Relevant log output

_No response_</details>

**Comment by sushreebarsa** [comment]
@jhogg11 
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999)
Thank you!

**Comment by jhogg11** [comment]
Okay, thanks for the response.

**Comment by sushreebarsa** [comment]
@jhogg11 Could you please confirm if you have posted the issue on  [keras-team/keras repo.](https://github.com/keras-team/keras/issues). Thank you!

**Comment by jhogg11** [comment]
> @jhogg11 Could you please confirm if you have posted the issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues). Thank you!

The Keras GitHub states to post it on the Tensorflow forum, and I did that, so that's one more venue where I've posted the question without a meaningful response.

https://discuss.tensorflow.org/t/basic-keras-model-underperforming-against-scikit-learn-mlpregressor/10485

**Comment by gadagashwini** [comment]
@gowthamkpr, I could able to reproduce the issue with `Tf 2.9`take a look at gist [here](https://colab.research.google.com/drive/1isnRSel2XfX3_pl5r1iogJSnVfiJtu6j?usp=sharing). Thank you!

**Comment by jhogg11** [comment]
> @gowthamkpr, I could able to reproduce the issue with `Tf 2.9`take a look at gist [here](https://colab.research.google.com/drive/1isnRSel2XfX3_pl5r1iogJSnVfiJtu6j?usp=sharing). Thank you!

Thanks for posting this.  By the way, if the loop is changed from:

`for _ in range(100):`

to

`for i in range(100):`

And then the value of `i` is used as the seed, including adding `random_state=i` to `make_regression`, that should give a fully consistent result every time.  Sorry, I missed that the first time.

When I run it like that, I get averages over 100 loops of:

```
SK 350.9508281761801
TF 372.78219372732224
```

So it gives a little bigger difference between the two than your result (and slightly smaller than what I originally posted).

And just to clarify, my assumption with this has been that there is something in the construction of the `MLPRegressor` model that should be able to be replicated in TF (rather than a bug or otherwise in TF).  I just can't figure out what it is.

**Comment by gowthamkpr** [comment]
@jhogg11 As mentioned [here](https://github.com/keras-team/keras/issues/16495#issuecomment-1150414919), KerasRegressor is deprecated and we are no longer maintaining/bug fixing it. Please forward this issue to https://github.com/adriangb/scikeras
